### PR TITLE
fix: support cloudflare workers

### LINKED ISSUE #366: The `credentials` field on `RequestInitializerDict` is not implemented; Cloudflare Workers issue.
I am experiencing the same issue with credentials that prevents `ky` from being used with Cloudflare Workers that `redaxios` is also [experiencing here](https://github.com/developit/redaxios/issues/64).

### PR DESCRIPTION
**Pull request by riderx**
This PR comes to fix the Cloudflare workers issue mentioned here with the solution:
https://github.com/sindresorhus/ky/issues/366#issuecomment-1670435774

Fixes #366

### CODE REVIEW

**Comment by sindresorhus** [comment]
CI is failing

**Comment by riderx** [comment]
@sindresorhus fixed :)

File: source/core/Ky.ts [hunk 1]
```diff
@@ -118,9 +118,10 @@ export class Ky {
 	// eslint-disable-next-line complexity
 	constructor(input: Input, options: Options = {}) {
 		this._input = input;
+		const isCredentialsSupported = 'credentials' in Request.prototype;
 		this._options = {
 			// TODO: credentials can be removed when the spec change is implemented in all browsers. Context: https://www.chromestatus.com/feature/4539473312350208
-			credentials: (this._input as Request).credentials || 'same-origin',
+			credentials: isCredentialsSupported ? ((this._input as Request).credentials || 'same-origin') : 'same-origin',
```
Comment (fregante) [inline source/core/Ky.ts:124]: I don’t think this is gonna work, `_options.credentials` should not be set at all, according to the error.

Also the comment is very outdated and we can probably use:

```suggestion
			credentials: isCredentialsSupported ? (this._input as Request).credentials : undefined,
```

… as long as CloudFlare accepts `{credentials: undefined}` and doesn't actually require `{}`

**Review by fregante — CHANGES_REQUESTED** [review]

**Comment by riderx** [comment]
Thanks @fregante i did update it.
I had to change the typing to make this solution work, and added a comment.
I did try and `{credentials: undefined}` is the required way for Cloudflare.
Since isCredentialsSupported is not reused, I wonder if that shouldn't be better, directly in the ternary?

**Review by fregante — APPROVED** [review]

**Pull request merged**